scripts/install_xcpng.py

```diff
@@ -62,24 +62,58 @@ def generate_answerfile(directory, installer, hostname_or_ip, type, hdd):
         elif type == 'upgrade':
             answerfile.write(f"""<?xml version="1.0"?>
 <installation mode="upgrade">
-    <existing-installation>sda</existing-installation>
+    <existing-installation>{hdd}</existing-installation>
     <source type="url">{installer}</source>
     <script stage="filesystem-populated" type="url">
         http://{pxe}/configs/presets/scripts/filesystem-populated.py
     </script>
 </installation>
         """)
+        elif type == 'restore':
+            answerfile.write(f"""<?xml version="1.0"?>
+<restore>
+    <existing-installation>{hdd}</existing-installation>
```

Comment:
Does this line do anything concretely? It's specific to the `upgrade` operation so I doubt it. I suppose it's ignored and the restore process auto-detects the existing installation.